etc/dev/curated_example.yml

```diff
@@ -0,0 +1,1007 @@
+# --- Common Meta-data for all PRIMARY HDUs
+PRIMARY_COMMON:
+ SIMPLE:
```

Comment:
As I said during our meeting, using dict keys here does not round trip properly. If I load this YAML file the values for `COMMENT` and `None` get overwritten such that you only get the final definitions. The order is retained because python >= 3.6 now gives you ordered dicts but that still does not match our goals. The header cards need to be stored in `PRIMARY_COMMON` as elements of a list where each element defines the header item.

Something like:

```yaml
PRIMARY_COMMON:
  - keyword: SIMPLE
    value: True
    comment: "A comment"
  - keyword: BITPIX
    value: 16
    comment: "Another comment"
```

Reply:
I've implemented the code to write the meta-data as a list structure instead of a dictionary. For now I'm letting yaml (via dump and load) to cast the values to the proper type.  I'd assume that this is the desired behavior. 
Therefore: 
- When a keyword in undef, such as the case of the blank comment  we get `null`
- For `COMMENTS` the keyword  is actually `COMMENTS`
- When a value is not defined, we get `value=null`
Now we can have multiple `COMMENTS` and blank lines. Order of the header is preserved too.